### PR TITLE
feat: migrate renameTable to typescript

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR migrates `renameTable` and `renameTableQuery` methods to typescript.
I have included a `changeSchema` option which has to be set in order to move a table from one schema to another if this is supported by the dialect.

### CODE REVIEW

File: packages/core/src/dialects/ibmi/query-generator-typescript.ts [hunk 1]
```diff
@@ -61,6 +62,21 @@ export class IBMiQueryGeneratorTypeScript extends AbstractQueryGenerator {
     ]);
   }
 
+  renameTableQuery(
+    beforeTableName: TableNameOrModel,
+    afterTableName: TableNameOrModel,
+    _options?: RenameTableQueryOptions,
```
Comment (WikiRik) [inline packages/core/src/dialects/ibmi/query-generator-typescript.ts:68]: We should add a rejectInvalidOptions with RENAME_TABLE_QUERY_SUPPORTABLE_OPTIONS so it's consistent with other methods

File: packages/core/test/unit/query-generator/rename-table-query.test.ts [hunk 2]
```diff
@@ -18,26 +21,34 @@ describe('QueryGenerator#renameTableQuery', () => {
...
 
-  it('produces a query that renames the table with schema', () => {
+  it('throws an error if `options.changeSchema` is not set when moving table to another schema', () => {
     expectsql(() => queryGenerator.renameTableQuery({ tableName: 'oldTable', schema: 'oldSchema' }, { tableName: 'newTable', schema: 'newSchema' }), {
-      default: 'ALTER TABLE [oldSchema].[oldTable] RENAME TO [newSchema].[newTable];',
-      mssql: 'EXEC sp_rename [oldSchema].[oldTable], [newSchema].[newTable];',
-      sqlite: 'ALTER TABLE `oldSchema.oldTable` RENAME TO `newSchema.newTable`;',
-      'db2 ibmi': 'RENAME TABLE "oldSchema"."oldTable" TO "newSchema"."newTable";',
+      default: changeSchemaNotSetError,
+      mssql: changeSchemaNotSetError,
```
Comment (WikiRik) [inline packages/core/test/unit/query-generator/rename-table-query.test.ts:33]:
```suggestion
```

File: packages/core/test/unit/query-generator/rename-table-query.test.ts [hunk 3]
```diff
@@ -1,15 +1,18 @@
 import { createSequelizeInstance, expectsql, sequelize } from '../../support';
 
 const dialect = sequelize.dialect;
+const changeSchemaNotSetError = new Error('To move a table between schemas, you must set `options.changeSchema` to true.');
+const moveSchemaNotSupportedError = new Error(`Moving tables between schemas is not supported by ${dialect.name} dialect.`);
```
Comment (WikiRik) [inline packages/core/test/unit/query-generator/rename-table-query.test.ts:5]: I think I'm missing a test that moves a table across schemas without a rename

File: packages/core/test/integration/query-interface/rename-table.test.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        expect(tableNames).to.not.contain('my_test_table');
+      });
+
+      it('should throw error when moving table to another schema if changeSchema is not set', async () => {
+        const promise = queryInterface.renameTable(
+          { tableName: 'my_test_table', schema },
+          { tableName: 'my_test_table', schema: dialect.getDefaultSchema() },
+        );
+
+        if (['db2', 'ibmi'].includes(dialect.name)) {
```
Comment (WikiRik) [inline packages/core/test/integration/query-interface/rename-table.test.ts:51]: Related to my comment about rejectInvalidOptions, use the DialectSupports from the dialects index.ts here. We'd like to limit hardcoding dialect names for when we split the dialects

**Review by WikiRik — COMMENTED** [review]
Looks good!

File: packages/core/test/unit/query-generator/rename-table-query.test.ts (hunk 3 above)
Comment (lohart13) [reply packages/core/test/unit/query-generator/rename-table-query.test.ts:6]: It did exist, I've renamed it now.

Comment (WikiRik) [reply packages/core/test/unit/query-generator/rename-table-query.test.ts:6]: I was looking for a test like
`queryGenerator.renameTableQuery({ tableName: 'oldTable', schema: 'oldSchema' }, { tableName: 'oldTable', schema: 'newSchema' }, { changeSchema: true })` (note that tableName is the same, only schema has changed)

Does that also exist?

Comment (lohart13) [reply packages/core/test/unit/query-generator/rename-table-query.test.ts:6]: I've added a new test now.
LMK if this is what you are looking for.

**Review by WikiRik — APPROVED** [review]

**Pull request merged**